pymc/model.py

```diff
@@ -1627,9 +1632,43 @@ def compile_fn(
             )
 
         if point_fn:
+            warnings.warn(
+                "Model.compile_fn(..., point_fn=True) is deprecated. Use `Model.compile_point_fn(...)` instead.",
+                DeprecationWarning,
+            )
```

Comment:
Why not start deprecating away from it altogether?

```suggestion
            warnings.warn(
                "Model.compile_fn(..., point_fn=True) is deprecated. In a future release it will no longer be possible to obtain a PointFunc.
                FutureWarning,
            )
```

And instead of creating the new verbose methods, let's just make sure they are not used internally (which you don't have to do on your own of course!)

Reply:
Or if it's too fast we should start adding a `point_fn` kwarg to the user facing functions (compile_logp, and so on), so that we can warn about the future breaking change and allow users to start setting it to False to disable the warning and have their code ready for when we finally remove the option to return a PointFunc. :/